src/System.Text.Encodings.CodePages/src/System/Text/DecoderBestFitFallback.cs

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+//
+// This is used internally to create best fit behavior as per the original windows best fit behavior.
+//
+
+using System;
+using System.Text;
+using System.Threading;
+using System.Diagnostics.Contracts;
+
+namespace System.Text
+{
+    internal sealed class InternalDecoderBestFitFallback : DecoderFallback
+    {
+        // Our variables
+        internal BaseCodePageEncoding encoding = null;
+        internal char[] arrayBestFit = null;
+        internal char cReplacement = '?';
+
+        internal InternalDecoderBestFitFallback(BaseCodePageEncoding encoding)
+        {
+            // Need to load our replacement characters table.
+            this.encoding = encoding;
```

Comment:
was the codeformatter run on this?; curious why we still have "this.encoding" vs _encoding = encoding;